### PR TITLE
[Ready]Logs when speech is forced by an external factor

### LINKED ISSUE #39931: Logging for nanites "Forced Speech" program
Currently the Investigate log for Nanites shows that someone has uploaded a "Forced Speech" program to the cloud but doesn't log what speech was included as part of the program.

As there's no easy way to differentiate nanite speech from real speech in logs, this is fairly important to have amended.

### LINKED ISSUE #39932: Logging for speech leaves out crucial information
Presently when a person is forced to say something (whether by brain damage or nanites or other) this shows up in logs exactly the same as normal speech.

This creates an admin issue where people can be made to say potentially rule breaking stuff (a lot of brain damage lines are minor or major OOC in IC, Nanites can be used to make people read WGW, etc) and from a logs perspective its impossible for the admin to distinguish these lines from normal speech, let alone notice the person has nanites or brain damage to cause this.

Ideally any kind of involuntary speech would be tagged as such in the in game logs as well as the public logs, preferably with the cause of the forced speech as well.

### PR DESCRIPTION
**Pull request by XDTM**
Adding a `forced` argument in a say() proc will log it as FORCED BY [forced]. Nanites and vodoo dolls should no longer be able to get people in trouble.

Fixes #39931
Fixes #39932

### CODE REVIEW

File: code/modules/mob/living/say.dm [hunk 1]
```diff
@@ -81,7 +81,7 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 
 	return new_msg
 
-/mob/living/say(message, bubble_type,var/list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE)
+/mob/living/say(message, bubble_type,var/list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null)
```
Comment (AnturK) [inline code/modules/mob/living/say.dm]: Needs to be added to any overrides too.

**Comment by coiax** [comment]
On a semi-serious note, what if an admin forces a say proc call? I know that proc calls are also logged, but not in the same place.

**Comment by XDTM** [comment]
The admin forcesay verb is logged as forced, calling say from vv isn't, but calling the proc itself is still traceable.

**Comment by Arianya** [comment]
Generally speaking, policy has been not to forcesay/proccall say on someone unless they would appreciate the joke (or its part of an event), and proc calls are flagged to online admins so it shouldn't cause issues related to #39932

**Pull request merged**